internal/output/drone/service.go

```diff
@@ -7,10 +7,16 @@ package drone
 import "github.com/drone/drone-yaml/yaml"
 
 // Service appends a new service.
-func (o *Output) Service(spec *yaml.Container) *Output {
-	spec.Volumes = append(spec.Volumes, o.standardMounts...)
+func (o *Output) Service(spec *yaml.Container) {
+	o.appendService(spec, o.defaultPipeline)
+}
 
-	o.defaultPipeline.Services = append(o.defaultPipeline.Services, spec)
+func (o *Output) appendService(originalService *yaml.Container, pipeline *yaml.Pipeline) {
+	// perform a shallow copy of the step to avoid modifying the original
+	spec := *originalService
+
+	spec.Volumes = append([]*yaml.VolumeMount{}, spec.Volumes...)
```

Comment:
We really [should](https://pkg.go.dev/golang.org/x/exp/slices#Clone) get `slices.Clone(slc)`